libafl/src/bolts/staterestore.rs

```diff
@@ -67,6 +67,15 @@ where
     phantom: PhantomData<*const SP>,
 }
 
+impl<SP> Drop for StateRestorer<SP>
+where
+    SP: ShMemProvider,
+{
+    fn drop(&mut self) {
+        println!("aaa");
```

Comment:
I don't know why but this drop() is not working

I don't see "aaa" printed when I press ctrl-c, but I see "bbb" so the handler is working

Reply:
so I want to release the self.shmem here. because it's never freed atm

Reply:
do you have any ideas why drop is not being called? @domenukk

Reply:
Maybe look at it in assembly, I would assume that the actual call to drop happens after sys::exit?

Reply:
it's not there in asm code either!

Reply:
I checked the asm code but still, it looks as if it is optimized out (but i'm on debug build)